assets/source/setup-guide/app/components/SaveSettingsButton.js

```diff
@@ -12,13 +13,55 @@ import {
 	useSettingsDispatch,
 	useCreateNotice,
 } from '../helpers/effects';
+import prepareForTracking from '../helpers/prepare-for-tracking';
 
+/**
+ * Clicking on "… Save changes" button.
+ *
+ * @event wcadmin_pfw_save_changes_button_click
+ *
+ * @property {boolean} enable_debug_logging Indicates if Enable debug logging option is checked
+ * @property {boolean} enhanced_match_support Indicates if Enhanced Match Support option is checked
+ * @property {boolean} erase_plugin_data Indicates if Erase Plugin Data option is checked
+ * @property {boolean} product_sync_enabled Indicates if Enable Product Sync option is checked
+ * @property {boolean} rich_pins_on_posts Indicates if Add Rich Pins for Posts option is checked
+ * @property {boolean} rich_pins_on_products Indicates if Add Rich Pins for Products option is checked
+ * @property {boolean} save_to_pinterest Indicates if Save to Pinterest option is checked
+ * @property {boolean} track_conversions Indicates if Track Conversion option is checked
+ * @property {string} context The context in which the event is recorded
+ */
+
+/**
+ * Save Settings button component
+ *
+ * @fires wcadmin_pfw_save_changes_button_click with
+ * ```
+ * {
+ * 	enable_debug_logging: true|false,
+ *	enhanced_match_support: true|false,
+ *	erase_plugin_data: true|false,
+ *	product_sync_enabled: true|false,
+ *	rich_pins_on_posts: true|false,
+ *	rich_pins_on_products: true|false,
+ *	save_to_pinterest: true|false,
+ *	track_conversions: true|false,
+ * 	context: 'pinterest_settings'
+ * }
+ * ```
+ * @return {JSX.Element} Rendered element
```

Comment:
:nail_care: 

I'd shorten it to 
```suggestion
 * @fires wcadmin_pfw_save_changes_button_click with `{ context: 'pinterest_settings', … }`
 *
 * @return {JSX.Element} Rendered element
```

As putting every other property with actually unknown value, IMHO blurs the picture. I'd start to think- are those all props, or is it a subset of event props? is any of them set to a specific value?

So instead of 
![image](https://user-images.githubusercontent.com/17435/146531238-7a0e96e2-78e2-4819-872a-3581563c0b59.png)

we'd get
![image](https://user-images.githubusercontent.com/17435/146531046-b7c0cccd-5bc9-40c5-a30a-31154c4db9d3.png)